synapse/lib/storm.py

```diff
@@ -2360,9 +2430,9 @@ async def initSubRuntime(self, query, opts=None):
         Construct and return sub-runtime with a shared scope.
         ( caller must fini )
         '''
-        snap = await self._snapFromOpts(opts)
+        view = await self._viewFromOpts(opts)
 
-        runt = await Runtime.anit(query, snap, user=self.user, opts=opts, root=self)
+        runt = await Runtime.anit(query, view, user=self.user, opts=opts, root=self, bus=self.bus)
```

Comment:
in view.exec we call `getSubRuntime` which would end up creating a linked runtime, similar to the `runas` use. This is a notable change in behavior.

Reply:
Yep, I added this to address a couple other bugs we had recorded, SYN-3488 for the view.exec usage and SYN-2354 for sub-runtimes in general.

Reply:
Updated to only link specific events in the view.exec and runas commands.